bin/check_hcid.py

```diff
@@ -76,37 +75,42 @@ def check_report_for_hcid(hcid_dict, taxonomy_dir, kreport_file):
             hcid_dict[taxid]["classified_parent_found"] = True
 
 
-def map_to_refs(query, reference, preset):
+def map_to_refs(query, ref_sam):
     counts = defaultdict(int)
     ranges = defaultdict(list)
     read_ids = defaultdict(list)
-    a = mp.Aligner(reference, best_n=1, preset=preset)  # load or build index
-    if not a:
-        raise Exception("ERROR: failed to load/build index")
+
+    in_file = open(ref_sam, 'r')
+    in_sam = Reader(in_file)
 
     read_count = 0
-    for name, seq, qual in mp.fastx_read(query):  # read a fasta/q sequence
+    y = None
+    for x in in_sam:
+        if x.flag not in [0,16]:
```

Comment:
Not sure if I'm looking in the wrong place but this attr isn't listed in the simplesam docs, I assume you're just limiting this to properly aligned reads?

Could this be done in a way which makes the logic more explicit?

Reply:
The flag attribute has an example on https://github.com/mdshw5/simplesam but does not appear to be in the docs. I am wanting to exclude secondary mappings etc. Especially where there are close relatives in the reference fasta file this seems to be necessary. There is no more explicit way to access this info in the sam (other than renaming/commenting within the code which I can do) and the minimap command previously already includes the flags `--secondary=no -N 1 --sam-hit-only` to avoid secondary, only provide top hit and not write unaligned reads to file so this step is more of a sanity check (it made a difference to getting the correct warnings before the refs were updated)

Reply:
I have checked and the `x.flag not in [0,16]` makes the difference between E. coli test case getting a Y. pestis warning or not. I've updated code with clearer variable names otherwise